modules/internal/String.chpl

```diff
@@ -427,6 +427,75 @@ module String {
       return ret;
     }
 
+    /* This function takes an array of stings as input.
+
+    returns:  The longest common prefix path of all the input paths
+    */
+    proc commonPrefix(paths: []): string {
+
+    var prefix: string = "";    // result string
+    var inputLength = paths.size;   // size of input array
+    var firstPath = paths[1];
+    var delimiter: string;
+    var cnt: int = 1;
+    var pos: int ;    // rightmost index of common prefix
+    var minPathLength: int ;
+
+    // if input is empty, return empty string.
+    // if input is just one string, return that string as longest common prefix path.
+
+    if inputLength == 0 then {
+      return prefix;
+    } else if inputLength == 1 then{
+      return firstPath;
+    }
+
+    // finding delimiter to split the paths.
+
+    if firstPath.find("\\", 1..firstPath.length) == 0 then {
+      delimiter = "/";
+    } else {
+      delimiter = "\\";
+    }
+    cnt = firstPath.count(delimiter);
+
+    var prefixArray:[1..cnt+1] string;    // array of resultant prefix string
+    prefixArray = firstPath.split(delimiter, -1, false);
+
+    pos = prefixArray.size;
+    minPathLength = prefixArray.size;
+
+    for i in 2..inputLength do {
+
+      cnt = paths[i].count(delimiter);
+      var tempArray:[1..cnt+1] string;    // temporary array storing the current path under consideration
+      tempArray = paths[i].split(delimiter, -1, false);
+
+      var minimum = min(prefixArray.size, cnt+1);
+      if minimum < minPathLength then {
+        minPathLength = minimum;
+      }
+
+      for (itr, ind) in zip(1..minimum, 1..minimum) do {
+        if tempArray[itr]!=prefixArray[ind] then {
+          pos = ind;
+          break;
+        }
+      }
+    }
+
+    if pos < prefixArray.size then {
+      prefixArray.remove(pos..prefixArray.size);
+      prefixArray.push_back(" ");
```

Comment:
There's probably something I'm missing, but why are you adding a " " here?

Reply:
It's for making a valid path. By adding the " " at the end of the array, we can add the delimiter at that position while using the join function.